optimus_manager/checks.py

```diff
@@ -47,6 +47,18 @@ def get_active_renderer():
     else:
         return "integrated"
 
+def get_integrated_provider():
+
+    try:
+        provider = exec_bash("xrandr --listproviders")
+    except BashError as e:
+        raise CheckError("Cannot list xrandr provdiers : %s")
+
+    for line in re.findall("name:.*", provider):
+        for _p in line.split("name:")[1].split():
+            if _p == "AMD":
+                return line.split("name:")[1]
+
```

Comment:
I'd prefer if that function always returned the name of the integrated provider (as its name suggest), without making a special case for AMD. Currently the function does not handle the Intel case at all, it just reaches its end and returns `None`.

We could simply iterate each line return by `xrandr --listproviders` and return the first name that isn't `NVIDIA-0`. This would also remove the need to put a if..else block around `xrandr --setprovideroutputsource` in `do_xsetup()` .